### PR TITLE
Add match operations to reg_shm and pageg mobj

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
Rebased

File: core/arch/arm/mm/mobj.c [hunk 1]
```diff
@@ -385,12 +385,24 @@ static TEE_Result mobj_reg_shm_get_cattr(struct mobj *mobj __unused,
...
+	.matches = mobj_reg_shm_matches,
 	.free = mobj_reg_shm_free,
 };
 
+static bool mobj_reg_shm_matches(struct mobj *mobj __maybe_unused,
+				   enum buf_is_attr attr)
+{
+	assert(mobj->ops == &mobj_reg_shm_ops);
+
+	return attr == CORE_MEM_NON_SEC || attr == CORE_MEM_NSEC_SHM ||
```
Comment (etienne-lms) [inline core/arch/arm/mm/mobj.c]: `CORE_MEM_NSEC_SHM` objects do not belong to the registered SHM areas (as `CORE_MEM_REG_SHM` do) as they belong to the static pre-mapped shared memory. I wonder if this change may confuse shm identification when core is built without debug support.

Comment (jenswi-linaro) [inline core/arch/arm/mm/mobj.c]: OK, I'll remove `CORE_MEM_NSEC_SHM`

**Comment by jenswi-linaro** [comment]
Update

**Comment by etienne-lms** [comment]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by jenswi-linaro** [comment]
Tag applied

**Pull request merged**